### ISSUE TITLE
Merge Erlang into Gamma?

### ISSUE THREAD

**Issue by lindahua** [JuliaStats/Distributions.jl#399]
An Erlang distribution is essentially a Gamma distribution with an integer shape parameter. 

Currently, we have a definition like this:
```
immutable Erlang <: ContinuousUnivariateDistribution
    gammad::Gamma
    ...
end
```
and all methods on `Erlang` are simply delegated to `d.gammad`. 

I am wondering whether should imply have one type `Gamma` here, and have `Erlang` as a special constructor of a Gamma distribution, as:

```julia
function Erlang(α, θ)
    isinteger(α) || 
        throw(ArgumentError("Erlang shape parameter must be an integer."))
    Gamma(α, θ)
end
```

**Comment by johnmyleswhite** [comment]
I think it's fair to deprecate Erlang, but it should be separate from Gamma since the MLE for an Erlang isn't the MLE for a Gamma in general.

**Comment by andreasnoack** [comment]
@johnmyleswhite Good point, but is there then a contradiction? How can you deprecate and keep it separate at the same time? For the reason you mention, I think we should keep them separate.

**Comment by johnmyleswhite** [comment]
Oh, I think you could also deprecate because you concede that our support for `Erlang` is kind of broken and the separation isn't real yet. But without assuming that we'd never support `Erlang`, just that we haven't supported it yet.

**Comment by andreasnoack** [comment]
I think we a are just using the same underlying driver functions whenever possible. I believe this is similar to binomial using the beta distribution cdf (through Rmath).

**Comment by lindahua** [comment]
Ok, I think we can just consider `Erlang` and `Gamma` as two different types that rely on the same set of computing functions. Also, we can provide a `convert` function to convert from `Erlang` to `Gamma`.

**Comment by johnmyleswhite** [comment]
Works for me.

**Comment by lindahua** [comment]
implemented in 9c891f0
